### PR TITLE
feat: add ArgoCD infrastructure

### PR DESCRIPTION
**Pull request by dymbe**
Add ArgoCD infrastructure. Includes both an ArgoCD-Application, and Kubernetes-YAML to be deployed

### CODE REVIEW

File: argocd/argocd-app.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (kstigen) [inline argocd/argocd-app.yaml:6]: Is this file really needed? When seeing in context with the argo-file in the infrastructure repo, it seems like the argo-file in the infrastructure repo could point directly to the infrastructure-folder in this repo?

File: argocd/infrastructure/geocoder-api.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: apps/v1
```
Comment (kstigen) [inline argocd/infrastructure/geocoder-api.yaml]: Hmm..should we rename the folder to `deployment` instead of infrastructure?

File: argocd/infrastructure/geocoder-api.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+kind: Deployment
+metadata:
+  name: geocoder-api-deployment
```
Comment (kstigen) [inline argocd/infrastructure/geocoder-api.yaml]: What namespace will this application be installed in?

Comment (dymbe) [reply argocd/infrastructure/geocoder-api.yaml]: The `argocd`-namespace

File: argocd/argocd-app.yaml (hunk 1 above)
Comment (dymbe) [reply argocd/argocd-app.yaml:6]: It could, but then all resources would belong to a single app. Instead of having this:
image
We would only see the "apps"-app. We would then lose the overview of the health- and sync-status of specific apps. It would also make it a bit more difficult to sync only one app

Comment (dymbe) [reply argocd/argocd-app.yaml:6]: I am however ok with moving this file to the infrastructure-repo. My initial thought was that I should move application-specific code as close to the application-repo as possible. That being said, this file won't need much application-specific customization.

File: argocd/infrastructure/geocoder-api.yaml (hunk 3 above)
Comment (dymbe) [reply argocd/infrastructure/geocoder-api.yaml]: This seems to be best practice, although they have a guide for putting them in any other namespace:
https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/
They do warn that misconfiguration could lead to vulnerabilities though. We should therefore have a good reason for putting applications in a namespace other than `argocd` IMO.

File: argocd/argocd-app.yaml (hunk 1 above)
Comment (kstigen) [reply argocd/argocd-app.yaml:6]: Let's keep it the way it is now.

File: argocd/infrastructure/geocoder-api.yaml (hunk 3 above)
Comment (kstigen) [reply argocd/infrastructure/geocoder-api.yaml]: Aiaiai...I don't like the idea that all our applications must be in the same namespace in kubernetes. That will prevent us from separating applications and their priveleges.

See if we can work around this to be able to deploy applications/services to multiple namespaces.

File: argocd/argocd-app.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: argocd-apps-geocoder-api
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: geocoder-api
+  namespace: argocd-apps-geocoder-api
```
Comment (kstigen) [inline argocd/argocd-app.yaml]: A better prefix without `argocd`

**Review by kstigen — DISMISSED** [review]

File: argocd/infrastructure/geocoder-api.yaml (hunk 3 above)
Comment (dymbe) [reply argocd/infrastructure/geocoder-api.yaml]: Fixed by https://github.com/openearthplatforminitiative/infrastructure/pull/5
